server/athenian/api/auth.py

```diff
@@ -0,0 +1,119 @@
+import logging
+import os
+import re
+import json
+import aiohttp.web
+from urllib.request import urlopen
+from urllib.request import Request as HTTPRequest
+from jose import jwt
+
+# Format error response and append status code.
+class AuthError(Exception):
+    def __init__(self, error, status_code):
+        self.error = error
+        self.status_code = status_code
+
+class User():
+    def __init__(self, info):
+        self.username = info["nickname"]
+        self.fullname = info["name"]
+        self.picture = info["picture"]
+        self.updated = info["updated_at"]
+
+AUTH0_DOMAIN = os.environ.get("AUTH0_DOMAIN")
+AUTH0_AUDIENCE = os.environ.get("AUTH0_AUDIENCE")
+
+def AuthConfigured():
+    return AUTH0_DOMAIN and AUTH0_AUDIENCE
+
+class Auth0():
+    def __init__(self, domain=AUTH0_DOMAIN, audience=AUTH0_AUDIENCE, whitelist=tuple()):
+        self._domain = domain
+        self._audience = audience
+        self._algos = ["RS256"]
+        self._whitelist = whitelist
+
+    def _check_request(self, request):
+        for pattern in self._whitelist:
+            if re.match(pattern, request.path):
+                return True
+        return False
+
+    def _get_token_auth_header(self, request):
```

Comment:
```suggestion
    def _get_token_auth_header(self, request: aiohttp.web.Request) -> str:
```
(nicer IDE experience)